pyproject.toml

```diff
@@ -105,6 +101,94 @@ test = [
     "coverage",
 ]
 
+[tool.ruff]
+target-version = "py38"  # minimum supported version
+line-length = 88  # same as Black.
+select = [
+    "A",  # flake8-builtins
+    "ANN",  # flake8-annotations
+    "ARG",  # flake8-unused-arguments
+    "B",  # flake8-bugbear
+    "BLE",  # flake8-blind-except
+    "C4",  # flake8-comprehensions
+    "D",  # pydocstyle
+    "E",  # pycodestyle error
+    "EM",  # flake8-errmsg
+    "ERA",  # eradicate
+    "F",  # Pyflakes
+    "FA",  # flake8-future-annotations
```

Comment:
I personally like this ruleset, but because we support 3.8 we should probably leave it out.  Some of the rules conflict with 3.8 and 3.9.  Like the use of `Optional` vs `|` in annotations.

Reply:
Setting `target-version = "py38"` sets our "minimum version," so all these upgrade and do-it-more-newer tools will only suggest changes that 3.8 supports.